sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StringOffsetMethods.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class StringOffsetMethods
+    {
+        public void StringOffsetMethodsCases()

```

Comment:
please also add tests with

- a variable of type string `x.Substring(1).IndexOf('t')`
- using `x.Substring(int startIndex, int length);`
- using other overloads of IndexOf chained to the `Substring` call
- using overloads with `StringComparison` as well

also, I would add some Noncompliant tests where this gets triggered inside a Constructor, a Getter and a lambda

also, I would add a test like
`x.Substring(1).Remove(1).IndexOf('t')`
and
`x.Remove(1).IndexOf('t')`

to document the behavior in these cases